rust/hyperlane-base/src/types/gcs_storage.rs

```diff
@@ -0,0 +1,214 @@
+use async_trait::async_trait;
+use derive_new::new;
+use eyre::Result;
+use hyperlane_core::{SignedAnnouncement, SignedCheckpointWithMessageId};
+use prometheus::IntGauge;
+use std::time::Duration;
+use std::{fmt, sync::OnceLock};
+use tokio::time::timeout;
+
+use crate::{settings::gcp_client::GoogleCloudClientProvider, CheckpointSyncer};
+
+const GCS_REQUEST_TIMEOUT_SECONDS: u64 = 30;
+
+#[derive(Clone, new)]
+/// Google Cloud Storage (GCS) type.
+pub struct GCSStorage {
+    /// The name of the bucket.
+    bucket: String,
+
+    /// A specific folder inside the above repo - set to empty string to use the root of the bucket
+    folder: Option<String>,
+
+    /// The region of the bucket.
+    region: String,
+
+    /// A client with GCS credentials.
+    #[new(default)]
+    authenticated_client: OnceLock<GoogleCloudClientProvider>,
```

Comment:
seems like there's no way to know how to set credentials

Reply:
Ah yes I'll add comments